roles/pulp/tasks/main.yml

```diff
@@ -14,6 +10,28 @@
     var: pulp_install_plugins_normalized
     verbosity: 1
 
+- name: Check if required variables are set
+  assert:
+    that:
+      # These checks run it through jinja2 templating twice: built-in
+      # "that:" (like when:), and then the {{ }}
+      # It converts from item, to the string like pulp_default_admin_password ,
+      # to the value of pulp_default_admin_password .
+      - "{{ item }} is not none"
+      - "{{ item }} | default('') | length > 0"
+    fail_msg: >
+      {{ item }} is undefined, null, an empty string, or an empty dictionary "{}" . Please
+      set it in your variable
+      (e.g. pulp_installer/playbooks/example-use/group_vars/all)
+      and run pulp_installer again.
+      See https://pulp-installer.readthedocs.io/en/latest/ or
```

Comment:
I like how helpful this error message is. :+1:

Reply:
FYI, it is all outputted as one line though.

I found instructions for the debug module (rather than assert) module to do multiple lines via an array, but it just becomes an array outputted as JSON, still one line.